modules/aws/auth.go

```diff
@@ -29,9 +30,25 @@ func NewAuthenticatedSession(region string) (*session.Session, error) {
 }
 
 // NewAuthenticatedSessionFromDefaultCredentials gets an AWS Session, checking that the user has credentials properly configured in their environment.
+// if TERRATEST_LOCALSTACK environment variable is set, uses LocalStack Endpoint and Credentials.
 func NewAuthenticatedSessionFromDefaultCredentials(region string) (*session.Session, error) {
 	awsConfig := aws.NewConfig().WithRegion(region)
 
+	if localStackUrl, ok := os.LookupEnv(LocalStackEnvVar); ok {
+		awsAccessKeyId := "test"
+		awsSecretAccessKey := "test"
+
+		if AWS_ACCESS_KEY_ID, ok := os.LookupEnv("AWS_ACCESS_KEY_ID"); ok {
+			awsAccessKeyId = AWS_ACCESS_KEY_ID
+		}
+
+		if AWS_SECRET_ACCESS_KEY, ok := os.LookupEnv("AWS_SECRET_ACCESS_KEY"); ok {
+			awsSecretAccessKey = AWS_SECRET_ACCESS_KEY
+		}
+
+		awsConfig = awsConfig.WithEndpoint(localStackUrl).WithDisableSSL(true).WithCredentials(credentials.NewStaticCredentials(awsAccessKeyId, awsSecretAccessKey, ""))
```

Comment:
Hi, will be helpful to have a test case to keep track that handling of `TERRATEST_LOCALSTACK` will continue to work

Reply:
unfortunately, that would require a lot more than you think as there aren't any tests for this for good reason.  I don't think there is a reasonable way to invoke the auth to begin with